mozilla-services/consvc-shepherd
# [SHEP-41] un-skip and fix test_upsert_deals in the boostr sync script tests

## References

[SHEP-41](https://mozilla-hub.atlassian.net/browse/SHEP-41)

## Problem Statement

The test for `test_upsert_deals` in the sync script tests needs to be fixed and unskipped.

## Proposed Changes

1. Fixed the `test_upsert_deals` tests
2. Added a test for when the Boostr API returns an empty array

## Verification Steps

Run the `make test` command to check the tests


[SHEP-41]: https://mozilla-hub.atlassian.net/browse/SHEP-41?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
